nan.h

```diff
@@ -1845,10 +1845,21 @@ namespace Nan {
 NAN_INLINE v8::Local<v8::Value> NanEncode(
     const void *buf, size_t len, enum Nan::Encoding encoding = Nan::BINARY) {
 #if (NODE_MODULE_VERSION > 0x000B)
-  return node::Encode(
-      v8::Isolate::GetCurrent()
-    , buf, len
-    , static_cast<node::encoding>(encoding));
+  v8::Isolate* isolate = v8::Isolate::GetCurrent();
+  node::encoding node_enc = static_cast<node::encoding>(encoding);
+
+  if (encoding == Nan::UCS2) {
+    return node::Encode(
+        isolate
+      , reinterpret_cast<const uint16_t *>(buf)
+      , len);
```

Comment:
This will not work properly on joyent/node 0.11.16, it will encode with the default encoding=node::BINARY. It is currently only io.js that has broken out UCS 2 handling.